detection_rules/config.py

```diff
@@ -50,101 +52,109 @@ class ConfigFile:
-        testing = cls.TestConfigPath(
-            config=testing_data['config']
-        ) if testing_data else None
+        testing_data = obj.get("testing")
+        testing = cls.TestConfigPath(config=testing_data["config"]) if testing_data else None
 
         return cls(files=files, rule_dir=rule_dir, testing=testing)
 
 
 @dataclass
 class TestConfig:
     """Detection rules test config file"""
-    test_file: Optional[Path] = None
-    unit_tests: Optional[UnitTest] = None
-    rule_validation: Optional[RuleValidation] = None
+
+    test_file: Path | None = None
+    unit_tests: UnitTest | None = None
+    rule_validation: RuleValidation | None = None
 
     @classmethod
-    def from_dict(cls, test_file: Optional[Path] = None, unit_tests: Optional[dict] = None,
-                  rule_validation: Optional[dict] = None) -> 'TestConfig':
-        return cls(test_file=test_file or None, unit_tests=UnitTest(**unit_tests or {}),
-                   rule_validation=RuleValidation(**rule_validation or {}))
+    def from_dict(
+        cls,
+        test_file: Path | None = None,
+        unit_tests: dict[str, Any] | None = None,
+        rule_validation: dict[str, Any] | None = None,
+    ) -> "TestConfig":
+        return cls(
+            test_file=test_file or None,
+            unit_tests=UnitTest(**unit_tests or {}),
+            rule_validation=RuleValidation(**rule_validation or {}),
+        )
 
     @cached_property
     def all_tests(self):
         """Get the list of all test names."""
         return discover_tests()
 
-    def tests_by_patterns(self, *patterns: str) -> List[str]:
+    def tests_by_patterns(self, *patterns: str) -> list[str]:
         """Get the list of test names by patterns."""
-        tests = set()
+        tests: set[str] = set()
         for pattern in patterns:
             tests.update(list(fnmatch.filter(self.all_tests, pattern)))
         return sorted(tests)
 
     @staticmethod
-    def parse_out_patterns(names: List[str]) -> (List[str], List[str]):
+    def parse_out_patterns(names: list[str]) -> tuple[list[str], list[str]]:
         """Parse out test patterns from a list of test names."""
-        patterns = []
-        tests = []
+        patterns: list[str] = []
+        tests: list[str] = []
         for name in names:
-            if name.startswith('pattern:') and '*' in name:
-                patterns.append(name[len('pattern:'):])
+            if name.startswith("pattern:") and "*" in name:
+                patterns.append(name[len("pattern:") :])
             else:
                 tests.append(name)
         return patterns, tests
 
     @staticmethod
-    def format_tests(tests: List[str]) -> List[str]:
+    def format_tests(tests: list[str]) -> list[str]:
         """Format unit test names into expected format for direct calling."""
-        raw = [t.rsplit('.', maxsplit=2) for t in tests]
-        formatted = []
+        raw = [t.rsplit(".", maxsplit=2) for t in tests]
```

Comment:
```suggestion
        raw: list[str] = [t.rsplit(".", maxsplit=2) for t in tests]
```

Reply:
`pyright` is smart enough to figure out the type of `raw` there -- `rsplit()` returns clear `list[str]`